src/test/java/com/commercetools/sync/services/impl/ShoppingListServiceImplTest.java

```diff
@@ -0,0 +1,274 @@
+        assertThat(errors.keySet())
+            .containsExactly("Failed to create draft with key: 'null'. Reason: Draft key is blank!");
+        verify(shoppingListSyncOptions.getCtpClient(), times(0)).execute(any());
+    }
+
+    @Test
+    void createShoppingList_WithEmptyShoppingListKey_ShouldHaveEmptyOptionalAsAResult() {
+        //preparation
+        final SphereClient sphereClient = mock(SphereClient.class);
+        final ShoppingListDraft shoppingListDraft = mock(ShoppingListDraft.class);
+        final Map<String, Throwable> errors = new HashMap<>();
+        when(shoppingListDraft.getKey()).thenReturn("");
+
+        final ShoppingListSyncOptions options = ShoppingListSyncOptionsBuilder
+            .of(sphereClient)
+            .errorCallback((exception, oldResource, newResource, actions) ->
+                    errors.put(exception.getMessage(), exception))
+            .build();
+
+        final ShoppingListService shoppingListService = new ShoppingListServiceImpl(options);
+
+        // test
+        final CompletionStage<Optional<ShoppingList>> result = shoppingListService
+            .createShoppingList(shoppingListDraft);
+
+        // assertion
+        assertThat(result).isCompletedWithValue(Optional.empty());
+        assertThat(errors.keySet())
+            .containsExactly("Failed to create draft with key: ''. Reason: Draft key is blank!");
+        verify(options.getCtpClient(), times(0)).execute(any());
+    }
+
+    @Test
+    void createShoppingList_WithUnsuccessfulMockCtpResponse_ShouldNotCreateShoppingList() {
+        // preparation
+        final ShoppingListDraft shoppingListDraft = mock(ShoppingListDraft.class);
+        final Map<String, Throwable> errors = new HashMap<>();
+        when(shoppingListDraft.getKey()).thenReturn("key");
+
+        final ShoppingListSyncOptions options = ShoppingListSyncOptionsBuilder
+                .of(mock(SphereClient.class))
+                .errorCallback((exception, oldResource, newResource, actions) ->
+                        errors.put(exception.getMessage(), exception))
+                .build();
+
+        final ShoppingListService shoppingListService = new ShoppingListServiceImpl(options);
+
+        when(options.getCtpClient().execute(any()))
+                .thenReturn(CompletableFutureUtils.failed(new InternalServerErrorException()));
+
+        // test
+        final CompletionStage<Optional<ShoppingList>> result =
+                shoppingListService.createShoppingList(shoppingListDraft);
+
+        // assertions
+        assertThat(result).isCompletedWithValue(Optional.empty());
+        assertThat(errors.keySet())
+                .hasSize(1)
+                .hasOnlyOneElementSatisfying(message -> {
+                    assertThat(message).contains("Failed to create draft with key: 'key'.");
+                });
+
+        assertThat(errors.values())
+                .hasSize(1)
+                .hasOnlyOneElementSatisfying(exception -> {
```

Comment:
this method has been deprecated.
```suggestion
                .singleElement().satisfies(exception -> {
```